pkg/retriever/retriever_test.go

```diff
@@ -31,96 +31,148 @@ func TestFilterersEventList(t *testing.T) {
+	if err != nil {
+		t.Fatalf("Unable to setup the contracts: %v", err)
+	}
+
 	filterers := []model.ContractFilterers{
-		filterer.NewCivilTCRContractFilterers(common.HexToAddress(testTCRAddress)),
+		filterer.NewCivilTCRContractFilterers(contracts.CivilTcrAddr),
+		filterer.NewNewsroomContractFilterers(contracts.NewsroomAddr),
 	}
-	retrieve := retriever.NewEventRetriever(client, filterers)
-	model1, _ := model.NewEventFromContractEvent("ApplicationWhitelisted", "CivilTCRContract", common.HexToAddress(testTCRAddress),
-		testEvent1, utils.CurrentEpochSecsInInt64(), model.Filterer)
-	model2, _ := model.NewEventFromContractEvent("Application", "CivilTCRContract", common.HexToAddress(testTCRAddress), testEvent2,
-		utils.CurrentEpochSecsInInt64(), model.Watcher)
+	retriever := retriever.NewEventRetriever(contracts.Client, filterers)
+	return contracts, retriever, filterers
+}
+
+func setupAllEvents(t *testing.T, contracts *cutils.AllTestContracts) int {
+	expectedNumTCREvents := setupTCRContractEvents(t, contracts)
+	// NOTE(IS): Setting up Newsroom contract emits 3 events
+	expectedNumNewsroomEvents := 3
+	expectedNumNewsroomEvents += setupNewsroomContractEvents(t, contracts)
+	numEvents := expectedNumTCREvents + expectedNumNewsroomEvents
+	return numEvents
+}
+
+// TestFilterers tests that past events are being retrieved upon Retrieve() call
+func TestRetrieveMethod(t *testing.T) {
+	contracts, retriever, _ := setupTestRetriever(t)
+
+	numEvents := setupAllEvents(t, contracts)
+	err := retriever.Retrieve()
 	if err != nil {
-		t.Errorf("Error connecting to rinkeby: %v", err)
+		t.Errorf("Error retrieving events: %v", err)
+	}
+
+	pastEvents := retriever.PastEvents
+
+	if len(pastEvents) != numEvents {
+		t.Errorf("Should have collected %v events but collected %v", numEvents, len(pastEvents))
 	}
-	retrieve.PastEvents = append(retrieve.PastEvents, model1, model2)
-	err = retrieve.SortEventsByBlock()
+}
+
+// TestSorting tests that sorting by block number works
+func TestSorting(t *testing.T) {
+	contracts, retriever, _ := setupTestRetriever(t)
+
+	numEvents := setupAllEvents(t, contracts)
+	err := retriever.Retrieve()
 	if err != nil {
-		t.Error("Sorting didn't happen")
+		t.Errorf("Error retrieving events: %v", err)
 	}
+
+	pastEvents := retriever.PastEvents
+
+	if len(pastEvents) != numEvents {
+		t.Errorf("Should have collected %v events but collected %v", numEvents, len(pastEvents))
+	}
+
+	err = retriever.SortEventsByBlock()
+	if err != nil {
+		t.Errorf("Error sorting events: %v", err)
+	}
+	sortedEvents := retriever.PastEvents
+
+	blockNumbers := make([]int, numEvents)
+	for idx, event := range sortedEvents {
+		blockNumbers[idx] = int(event.BlockNumber())
+	}
+	if !sort.IntsAreSorted(blockNumbers) {
+		t.Error("Events are not sorted")
+	}
+
 }
 
-// Check last events. TODO: make better test here w simulated backend
-// XXX(PN): Infura has been having problems, so commenting this out so builds work.
-// func TestLastEvents(t *testing.T) {
```

Comment:
Same question as above, is this test invalid now?

Reply:
oops, I changed the name. It should be TestLastEvents() sry about that